gooddata-sdk/gooddata_sdk/catalog/export/service.py

```diff
@@ -0,0 +1,90 @@
+# (C) 2023 GoodData Corporation
+import time
+from pathlib import Path
+from typing import Callable, Union
+
+from gooddata_api_client.exceptions import NotFoundException
+from gooddata_api_client.model.pdf_export_request import PdfExportRequest
+from gooddata_api_client.model.tabular_export_request import TabularExportRequest
+from gooddata_sdk import ExportRequest, GoodDataApiClient
+
+
+class ExportService:
+    def __init__(self, api_client: GoodDataApiClient) -> None:
+        self._entities_api = api_client.entities_api
+        self._actions_api = api_client.actions_api
+
+    @staticmethod
+    def _get_exported_content(
+        workspace_id: str, export_id: str, get_func: Callable, timeout: int = 60, retry: int = 5
+    ) -> bytes:
+        assert timeout > 0 and retry > 0, f"Timeout value '{timeout}' or retry value '{retry}' is negative."
+        response = get_func(workspace_id=workspace_id, export_id=export_id, _preload_content=False)
+        if response.status == 202:
+            counter = 0
+            while counter * retry <= timeout:
+                time.sleep(retry)
```

Comment:
5 seconds is too much.
Let's start with let's say 200ms and double the retry time in each iteration until it reaches 5s, then keep 5s.